### PR TITLE
add majority element

### PR DESCRIPTION
**Pull request by shivamp1998**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: Data-Structures/Array/MajorityElement.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/**
+ * [Majority Element](https://www.geeksforgeeks.org/majority-element/) A majority element in an array A[] of size n is an element that appears more than n/2 times (and hence there is at most one such element)
+ * @complexity: O(n) (on average )
+ * @complexity: O(n) (worst case)
+ * @flow
+ */
+
+// To find the candidate for majority element
+function Candidate (array) {
```
Comment (appgurueu) [inline Data-Structures/Array/MajorityElement.js:9]: Please name this `findCandidate` and remove the comment.

File: Data-Structures/Array/MajorityElement.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  let indexMajority = 0
+  let count = 1
+  const size = array.length
+  for (let i = 1; i < size; i++) {
+    if (array[indexMajority] === array[i]) {
+      count += 1
```
Comment (appgurueu) [inline Data-Structures/Array/MajorityElement.js:15]: `++`

File: Data-Structures/Array/MajorityElement.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    } else {
+      count -= 1
```
Comment (appgurueu) [inline Data-Structures/Array/MajorityElement.js:17]: `--`

File: Data-Structures/Array/MajorityElement.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        indexMajority = i
+        count = 1
+      }
+    }
+  }
+  return array[indexMajority]
+}
+
+//  verifies if candidate occurs more than two times in an array
+function isMajority (array, candidate) {
```
Comment (appgurueu) [inline Data-Structures/Array/MajorityElement.js:28]: `isMajorityElement`

File: Data-Structures/Array/MajorityElement.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  let count = 0
+  const size = array.length
+  for (let i = 0; i < size; i++) {
+    if (array[i] === candidate) {
+      count += 1
```
Comment (appgurueu) [inline Data-Structures/Array/MajorityElement.js:33]: `++`

File: Data-Structures/Array/MajorityElement.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    }
+    if (count > size / 2) {
```
Comment (appgurueu) [inline Data-Structures/Array/MajorityElement.js:35]: You can move this after incrementing count (`if (++count > size/2) ...`)

File: Data-Structures/Array/MajorityElement.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+      return 1
```
Comment (appgurueu) [inline Data-Structures/Array/MajorityElement.js:36]: Please return bools

File: Data-Structures/Array/MajorityElement.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    }
+  }
+  return 0
```
Comment (appgurueu) [inline Data-Structures/Array/MajorityElement.js:39]: Same here

File: Data-Structures/Array/MajorityElement.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+function Majority (array) {
```
Comment (appgurueu) [inline Data-Structures/Array/MajorityElement.js:42]: `findMajorityElement`; this is not a class

File: Data-Structures/Array/MajorityElement.js [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+  //    finds the candidate for majority
```
Comment (appgurueu) [inline Data-Structures/Array/MajorityElement.js:43]: self-explanatory if you rename the function

File: Data-Structures/Array/MajorityElement.js [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+  const cand = Candidate(array)
+  //    checks if it is the majority element
```
Comment (appgurueu) [inline Data-Structures/Array/MajorityElement.js:45]: same here

File: Data-Structures/Array/MajorityElement.js [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+  if (isMajority(array, cand)) {
+    return cand
+  } else return -1
```
Comment (appgurueu) [inline Data-Structures/Array/MajorityElement.js:48]: You can use a ternary here

File: Data-Structures/Array/MajorityElement.js [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+const FindMajority = (array) => {
```
Comment (appgurueu) [inline Data-Structures/Array/MajorityElement.js:51]: Redundant wrapper

File: Data-Structures/Array/MajorityElement.js (hunk 1 above)
Comment (shivamp1998) [reply Data-Structures/Array/MajorityElement.js]: Done

File: Data-Structures/Array/MajorityElement.js (hunk 2 above)
Comment (shivamp1998) [reply Data-Structures/Array/MajorityElement.js]: Done

File: Data-Structures/Array/MajorityElement.js (hunk 3 above)
Comment (shivamp1998) [reply Data-Structures/Array/MajorityElement.js]: Done

File: Data-Structures/Array/MajorityElement.js (hunk 4 above)
Comment (shivamp1998) [reply Data-Structures/Array/MajorityElement.js]: Done

File: Data-Structures/Array/MajorityElement.js (hunk 5 above)
Comment (shivamp1998) [reply Data-Structures/Array/MajorityElement.js]: Done

File: Data-Structures/Array/MajorityElement.js (hunk 6 above)
Comment (shivamp1998) [reply Data-Structures/Array/MajorityElement.js]: Done

File: Data-Structures/Array/MajorityElement.js (hunk 7 above)
Comment (shivamp1998) [reply Data-Structures/Array/MajorityElement.js]: Done

File: Data-Structures/Array/MajorityElement.js (hunk 8 above)
Comment (shivamp1998) [reply Data-Structures/Array/MajorityElement.js]: Done

File: Data-Structures/Array/MajorityElement.js (hunk 9 above)
Comment (shivamp1998) [reply Data-Structures/Array/MajorityElement.js]: Done

File: Data-Structures/Array/MajorityElement.js (hunk 10 above)
Comment (shivamp1998) [reply Data-Structures/Array/MajorityElement.js:42]: Done

File: Data-Structures/Array/MajorityElement.js (hunk 11 above)
Comment (shivamp1998) [reply Data-Structures/Array/MajorityElement.js:44]: Done

File: Data-Structures/Array/MajorityElement.js (hunk 12 above)
Comment (shivamp1998) [reply Data-Structures/Array/MajorityElement.js]: Done

File: Data-Structures/Array/MajorityElement.js (hunk 13 above)
Comment (shivamp1998) [reply Data-Structures/Array/MajorityElement.js]: Done

File: Data-Structures/Array/test/MajorityElement.test.js [hunk 14]
```diff
@@ -0,0 +1,8 @@
+import { FindMajority } from '../MajorityElement'
+
+describe('Majority element cases', () => {
```
[the remark this answers is not in the document]
Comment (shivamp1998) [reply Data-Structures/Array/test/MajorityElement.test.js:3]: Added

**Comment by shivamp1998** [comment]
I dont know why other files are changing!

**Pull request closed without merging**